iothub/device/src/InternalClient.cs

```diff
@@ -27,41 +27,33 @@ internal class InternalClient : IDisposable
         private readonly HttpTransportHandler _fileUploadHttpTransportHandler;
         private readonly IotHubClientOptions _clientOptions;
 
-        // Stores message input names supported by the client module and their associated delegate.
-        private volatile Dictionary<string, Tuple<MessageHandler, object>> _receiveEventEndpoints;
-
-        private volatile Tuple<MessageHandler, object> _defaultEventCallback;
-
-        // Stores methods supported by the client device and their associated delegate.
-
-        private bool _isDeviceMethodEnabled;
-        private readonly Dictionary<string, Tuple<MethodCallback, object>> _deviceMethods = new();
-
-        private volatile Tuple<MethodCallback, object> _deviceDefaultMethodCallback;
-
-        private volatile ConnectionStateChangeHandler _connectionStateChangeHandler;
-
-        // Count of messages sent by the device/ module. This is used for sending diagnostic information.
-        private int _currentMessageCount;
-
-        private int _diagnosticSamplingPercentage;
-
+        // Connection state change information
+        private volatile Action<ConnectionState, ConnectionStateChangeReason> _connectionStateChangeHandler;
```

Comment:
This could be tackled in a later PR, but I'd like to see us create a bundle class for the information we provide in this callback. That way, we can easily add new fields like previousState, the exception that caused it, etc.

https://github.com/Azure/azure-iot-sdk-java/blob/main/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ConnectionStatusChangeContext.java

Reply:
+1. We'll keep that for another PR. We'll also want to take another look at all our callback args and see if there is any room for improvement in there.